elastic/search_result.go

```diff
@@ -146,6 +167,52 @@ func (h *Hit) DecodeDNS(cfg *SearchConfig) (*client.DNSEntry, error) {
 	return e, nil
 }
 
+// protoFromNum returns a protocol string derived from protoNum (ie. "tcp" or "udp"),
+// otherwise it returns a string representation of the protoNum itself.
+func protoFromNum(protoNum uint8) string {
+	switch protoNum {
+	case ProtoTCP:
+		return "tcp"
+	case ProtoUDP:
+		return "udp"
+	default:
+		// We don't know what it is... at least return the number so that we may choose
+		// to add a mapping for the protocol.
+		return strconv.Itoa(int(protoNum))
+	}
+}
+
+// getProtocol tries to obtain a protocol from an elastic document, returning its string
+// representation.  If the found protocol is numeric (ie. 6 or "6"), it will be converted
+// to a non-numeric string (ie. "tcp" or "udp") if known, or will be converted to a numeric
+// string (ie. "3") if not known.  If the found protocol is not numeric, it will be returned
+// as is (ie. "tcp" or "udp" or "foo").  An empty string indicates an error, or simply the
+// fact that no protocol was supplied.
+func getProtocol(h *Hit, protoFp FieldPath) string {
```

Comment:
I think refactoring this function to
```go
func (h *Hit) sourceProtocol(path []string) string
```
will make it more in line with the rest of the bunch (sourceString, sourceInt64, etc.)